File: GitDesktop/GitDesktop/Models/WorkflowPreferences.swift
import Foundation

/// Collection of configurable settings regarding how the user may work with a repository.
/// Direct port of `electron/app/src/models/workflow-preferences.ts`.
/// GitHub fork-contribution target is retained as inert data only;
/// there is no GitHub integration (see AGENTS.md scope).
public enum ForkContributionTarget: String, Codable, Sendable {
    case parent = "parent"
    case `self` = "self"
}

public struct WorkflowPreferences: Codable, Sendable, Equatable {
    public var forkContributionTarget: ForkContributionTarget?

    public init(forkContributionTarget: ForkContributionTarget? = nil) {
        self.forkContributionTarget = forkContributionTarget
    }
}

File: electron/app/src/models/workflow-preferences.ts
export enum ForkContributionTarget {
  Parent = 'parent',
  Self = 'self',
}

/**
 * Collection of configurable settings regarding how the user may work with a repository.
 */
export type WorkflowPreferences = {
  /**
   * What repo does the user want to contribute to with this fork?
   */
  readonly forkContributionTarget?: ForkContributionTarget
}
